pippy_app.py

```diff
@@ -855,7 +855,46 @@ def _get_current_code(self):
         except Exception as e:
             print(f"Error reading file {current_file}: {e}")
             return None
-    
+
+    def markdown_parser(self, md: str):
+        lines = md.splitlines()
+        output = []
+        in_code_block = False
+
+        for line in lines:
+            stripped = line.strip()
+
+            if stripped.startswith("```"):
+                in_code_block = not in_code_block
+                continue
+
+            if in_code_block:
+                output.append("\033[1m" + line + "\033[0m\r\n")
+                continue
+
+            if stripped.startswith("### "):
+                output.append("\033[1;34;4m" + stripped[4:] + "\033[0m\r\n")
+                continue
+            elif stripped.startswith("## "):
+                output.append("\033[1;34;4m" + stripped[3:] + "\033[0m\r\n")
+                continue
+            elif stripped.startswith("# "):
+                output.append("\033[1;34m;" + stripped[2:] + "\033[0m\r\n")
+                continue
+
+            if stripped.startswith("- "):
+                line = "• " + stripped[2:]
+
+            line = re.sub(r"`([^`]*)`", r"\033[1m\1\033[0m", line)
+
+            line = re.sub(r"\*\*(.*?)\*\*", r"\033[2m\1\033[0m", line)
+
+            line = re.sub(r"(?<!\*)\*(?!\*)(.*?)\*(?!\*)", r"\033[2m\1\033[0m", line)
```

Comment:
While these regexes are great, a comment for each telling what their target is would be awesome.